alibuild_helpers/utilities.py

```diff
@@ -383,18 +388,49 @@ def getPackageList(packages, specs, configDir, preferSystem, noSystem,
     except TypeError as e:
       dieOnError(True, "Malformed entry prefer_system: %s in %s" % (systemRE, spec["package"]))
     if not noSystem and (preferSystem or systemREMatches):
-      cmd = "REQUESTED_VERSION=%s\n" % resolve_version(spec, defaults, "unavailable", "unavailable")
-      cmd += spec.get("prefer_system_check", "false").strip()
-      if not spec["package"] in testCache:
-        testCache[spec["package"]] = performPreferCheck(spec, cmd.strip())
-
+      requested_version = resolve_version(spec, defaults, "unavailable", "unavailable")
+      cmd = dedent("""\
+      REQUESTED_VERSION={version}
+      alibuild_system_replace () {{
+        echo "alibuild_system_replace: $1"
+        exit 0
+      }}
```

Comment:
maybe I misexpressed myself. I think we do not need an helper for a printout. Let's just document it.

Reply:
Fine, removed!